opengrok-web/src/main/java/org/opengrok/web/api/v1/RestApp.java

```diff
@@ -33,7 +34,9 @@
     public static final String API_PATH = "/api/v1";
     
     public RestApp() {
+        System.setProperty("sun.net.http.allowRestrictedHeaders", "true"); // necessary to test CORS from controllers
```

Comment:
can't you set it somewhere it tests? so the production code is not poluted with this

Reply:
or like this https://stackoverflow.com/questions/32137121/system-setpropertysun-net-http-allowrestrictedheaders-true-is-not-working you can set it for the surefire argline.